contrib_bots/lib/CommuteBot/CommuteBot.py

```diff
@@ -0,0 +1,150 @@
+from __future__ import print_function
+import requests
+
+class CommuteHandler(object):
+    '''
+    This plugin provides information regarding commuting from an origin to a destination, providing
+    a multitude of information. It looks for messages starting with '@commute'.
+    '''
+
+    def usage(self):
+        return '''
+            This plugin will allow briefings of estimated travel times,
+              distances and fare information for transit travel. It can vary outputs
+              depending on traffic conditions, departure and arrival times as well as
+              user preferences (toll avoidance, preference for bus travel, etc.).
+            It looks for messages starting with '@commute'.
+
+            Users should input an origin, destination and a mode of transport.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        # This next line of code is defensive, as we
+        # never want to get into an infinite loop of posting follow
+        # ups for own follow ups!
+        if message['display_recipient'] == 'commute':
+            return False
+        is_commute = (original_content.startswith('@commute') or
+                        original_content.startswith('@commute'))
+
+        return is_commute
+
+    # allows bot to send info via private message or same stream depending on
+    # user input
+    def send_info(self, message, letter, client):
+        if message['type'] == 'private':
+            client.send_message(dict(
+                type='private',
+                to=message['sender_email'],
+                content=letter,
+            ))
+        else:
+            client.send_message(dict(
+                type='stream',
+                subject=message['subject'],
+                to=message['display_recipient'],
+                content=letter,
+            ))
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        content_list = original_content.split()
+
+        if "help" in content_list:
+            help_info = '''
+    Obligatory Inputs:
+        Origin (origin=New York,NY,USA)
+        Destination (destinations=Chicago,IL,USA)
+        Mode Of Transport (Accepted inputs:driving, walking, bicycling, transit) (mode=driving)
+    Optional Inputs:
+        Units (Metric or imperial) (units=metric)
+        Restrictions (Accepted inputs:tolls, highways, ferries, indoor) (avoid=tolls)
+        Departure Time (Seconds After Midnight January 1, 1970 UTC) (departure_time=now)
+        Arrival Time (Seconds After Midnight January 1, 1970 UTC) (arrival_time=732283981023)
```

Comment:
This is a very inconvenient way of inputting time - how can a user input something in seconds from a date?